### ISSUE TITLE
The "insert SQL" statement must be executed at compile time ?

### ISSUE THREAD

**Issue by zdzDesigner** [speed2exe/myzql#14]
[myzql/src/protocol/prepared_statements.zig](https://github.com/speed2exe/myzql/blob/main/src/protocol/prepared_statements.zig#L112)
```zig
comptime var enum_field_types: [params.len]constants.EnumFieldType = undefined;
```
i am want to insert data at runtime, but the `execute` function only supports comptime
```zig
pub const Stage = struct {
    id: u32 = 0,
    user_id: u32 = 0,
    stage_id: u32 = 0,
    update_time: u32 = 0,
    parent_id: u32,
    title: []const u8,
    data: []const u8,
    const Tuple = struct { u32, u32, u32, u32, []const u8, []const u8 };
    pub fn tableName() []const u8 {
        return "stage";
    }
    pub fn toTuple(self: Stage) Tuple {
        return Tuple{ self.user_id, self.stage_id, self.update_time, self.parent_id, self.title, self.data };
    }
};
...
...
const exe_res = try client.execute(&prep_stmt, stage.toTuple());
```

**Comment by speed2exe** [comment]
@zdzDesigner It does not mean that way. `enum_field_types` will be known at comptime and is derived from the type `struct { u32, u32, u32, u32, []const u8, []const u8 }` which you defined, but not the values of it. Are you having error related to comptime?

**Comment by zdzDesigner** [comment]
@speed2exe  yes, like this 
```sh
 error: unable to resolve comptime value
                enum_field_type.* = comptime enumFieldTypeFromParam(param);
```

**Comment by speed2exe** [comment]
@zdzDesigner May I know which version of zig and myzql are you using?

**Comment by zdzDesigner** [comment]
Of course!
- zig version  0.13.0
- myzql version 
```sh
commit 47c9d47fd3a59c78b737a9035f8ec262095e9ff4 (HEAD)
Author: Zack Fu Zi Xiang <speed2exe@live.com.sg>
Date:   Sat Jun 29 19:55:21 2024 +0800

    fix: integration test

diff --git a/.github/workflows/unit_test.yml b/.github/workflows/unit_test.yml
index dfbbbfa..28fdb8c 100644
--- a/.github/workflows/unit_test.yml
+++ b/.github/workflows/unit_test.yml
@@ -15,7 +15,7 @@ jobs:
       - uses: actions/checkout@v3
       - name: install zig
         run: |
-          ZIG_VERSION=0.12.0-dev.3291+17bad9f88
+          ZIG_VERSION=0.13
           wget https://ziglang.org/builds/zig-linux-x86_64-$ZIG_VERSION.tar.xz
           tar xf zig-linux-x86_64-$ZIG_VERSION.tar.xz
           mv zig-linux-x86_64-$ZIG_VERSION $HOME/zig-build

commit 5d79f3a90865ac83e84cf2daa1fa76ae6270bd36
Author: Zack Fu Zi Xiang <speed2exe@live.com.sg>
Date:   Sat Jun 29 19:53:31 2024 +0800

    chore: update to 0.13
```

**Comment by zdzDesigner** [comment]
Thank you! I tested v0.13.2, and the issue has been resolved 👍.

**Comment by speed2exe** [comment]
@zdzDesigner Thanks for reporting the issue, I have reproduced your issue, and it is fixed in the latest tag `0.13.2` for zig `0.13.0`. Do let me know if you have any issues. If everything goes well. Feel free to close the issue.
